game/src/main/scala/io/aigar/game/Cell.scala

```diff
@@ -15,8 +15,7 @@ object Cell {
   final val MinMass = 1 // TODO determine this once we have visualization/etc.
 }
 
-class Cell(id: Int, startPosition: Vector2 = new Vector2(0f, 0f)) {
-  var position = startPosition
+class Cell(id: Int, grid: Grid, startPosition: Vector2 = new Vector2(0f, 0f)) {
```

Comment:
How about passing it only on `update`? That's the only place where it has the dependency, really. `Game` would pass it to `Player.update`, which would pass it to `Cell.update`.

Another option that I thought of would be to have a `constrain` method or similar in `Grid` that would take a list of `Cell`s and limit the positions.